src/include/miopen/hipoc_kernel.hpp

```diff
@@ -56,11 +56,10 @@ struct KernelArgsPair
     static const int second_index = sizeof(T) + padding;
     KernelArgsPair(T x, U y)
     {
-
         new(buffer) T(x); // NOLINT (clang-analyzer-cplusplus.PlacementNew)
         new(buffer + second_index) U(y);
     }
-    char buffer[second_index + sizeof(U)] = {};
+    alignas(U) char buffer[second_index + sizeof(U)] = {};
```

Comment:
This may change the layout of kernel arguments in memory. Without alignas(U), the required `buffer` alignment is 1 (i.e., no alignment is required). Therefore, the alignment of KernelArgsPair instances is also 1, and several such instances reside in memory without any gaps.

With alignas(U), the padding required for KernelArgsPair is alginof(U), which may lead to gaps between instances of KernelArgsPair.

If you see kernel failures, then please revert this change.

Reply:
@atamazov I do see lots of failures on `gfx906` with
```
handlehip.cpp:80: Failed getting available memory: invalid argument
```
However, it looks more likely a runtime issue since it cannot be reproduced on other newer ASICs. But I will revert this change and try again.

Reply:
@junliume This is not related and I see the same on navi21.

Reverting this change won't resolve the issue with getting available memory.

Reply:
@junliume It seems like this change is indeed correct and should be kept.

Reply:
@atamazov can you verify it happens on Navi21?
For Vega nodes some have problems but some other nodes do not.

Reply:
@junliume Of course, this happens on Navi21, as I've reported a while ago at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2307#issuecomment-1675977376 where you can find the dirty hacks for this. I am working on more or less regular W/A which should be suitable for merging into develop.